### ISSUE TITLE
[sqlite] time grain isn't set properly

### ISSUE THREAD

**Issue by csc5k** [apache/incubator-superset#3746]
Make sure these boxes are checked before submitting your issue - thank you!

- [x ] I have checked the superset logs for python stacktraces and included it here as text if any
- [x ] I have reproduced the issue with at least the latest released version of superset
- [x ] I have checked the issue tracker for the same issue and I haven't found one similar

### Superset version
0.20.4

### Expected results
When editing a bar graph slice, specify a time grain of "week"
X-axis of the bar graph runs on a weekly interval

The same would be true of "month" granularity

### Actual results
When editing a bar graph slice, specify a time grain of "week"
X-axis of the bar graph remains on a daily interval

The same is true of "month" granularity

### Steps to reproduce
Specify a time grain of "week" for a bar graph slice
Run Query in slice editor or save slice and view dashboard

I tried manually editing the "time_grain_sqla" parameter in the slice editor to be set to "week," it caused no effect.

**Comment by xrmx** [comment]
Please open bugs only against latest release. A lot has been fixed since 0.19.0

**Comment by claycrosby** [comment]
OK. Can you let me know if this has been fixed in subsequent releases or point me to documentation that would provide this info?

**Comment by xrmx** [comment]
Sorry, but the burden on verifying it's fixed or not is on you. The point of reproducing with latest release is not to waste time to research stuff that may be already be fixed :)

**Comment by claycrosby** [comment]
Let me rephrase the second part of my post: where are the release notes?

**Comment by claycrosby** [comment]
I upgraded Superset and the problem still persists.

**Comment by claycrosby** [comment]
I upgraded Superset and the problem still persists. The x-axis does not change granularity as I adjust the time grain in the slice.

**Comment by mistercrunch** [comment]
Is this the "Time Series - Bar Chart" or the "Distribution - Bar Chart" ?

**Comment by claycrosby** [comment]
Time Series - Bar Chart

**Comment by mistercrunch** [comment]
Which database backend is this?

**Comment by claycrosby** [comment]
SQLite 3

On Mon, Nov 6, 2017 at 10:05 AM, Maxime Beauchemin <notifications@github.com

**Comment by mistercrunch** [comment]
I can't reproduce with the example data...

**Comment by claycrosby** [comment]
do you want to get on a Google hangout or something so I can show you what
I'm seeing? or I can send screenshots

On Mon, Nov 6, 2017 at 10:08 AM, Maxime Beauchemin <notifications@github.com

**Comment by claycrosby** [comment]
Daily time grain:
[image: Inline image 1]

Weekly time grain (I clicked Run Query to update it):
[image: Inline image 2]

I want to see the data aggregated for each week

**Comment by jkokatjuhha** [comment]
Get the same error. Checked the SQL and it seems that the parameters are set incorrectly all the time. 

Weekly:
```
SELECT DATE(cancel_time, -strftime('%%w', cancel_time) || ' days') AS __timestamp,
       COUNT(ordernumber) AS count
FROM canc_reasons
WHERE cancel_time >= '2017-11-07 00:00:00.000000'
  AND cancel_time <= '2017-11-14 20:17:56.000000'
GROUP BY DATE(cancel_time, -strftime('%%w', cancel_time) || ' days')
ORDER BY count DESC
LIMIT 50000
OFFSET 0
```

Monthly:
```
SELECT DATE(cancel_time, -strftime('%%d', cancel_time) || ' days', '+1 day') AS __timestamp,
       COUNT(ordernumber) AS count
FROM canc_reasons
WHERE cancel_time >= '2017-11-07 00:00:00.000000'
  AND cancel_time <= '2017-11-14 20:20:18.000000'
GROUP BY DATE(cancel_time, -strftime('%%d', cancel_time) || ' days', '+1 day')
ORDER BY count DESC
LIMIT 50000
OFFSET 0
```

**Comment by mistercrunch** [comment]
Here's where it's defined in the source if someone wants to tackle this.
https://github.com/apache/incubator-superset/blob/master/superset/db_engine_specs.py#L296

**Comment by tony0918** [comment]
Same issue for Clickhouse data source on version 0.22.1

SELECT toStartOfMinute(toDateTime(dt)) AS __timestamp,
       COUNT(*) AS number_of_requests
FROM abc
WHERE DateReceived >= toDateTime('2018-02-24 00:00:00')
  AND DateReceived <= toDateTime('2018-02-24 23:59:59')
  AND dt >= toDate('2018-02-24')
  AND dt <= toDate('2018-02-24')
GROUP BY toStartOfMinute(toDateTime(dt))
ORDER BY number_of_requests DESC
LIMIT 5000

**Comment by villebro** [comment]
Attn: @claycrosby @jkokatjuhha I am unable to to replicate this bug; have you tested against a fresh version? If it's still present I can work on this issue if someone can help me reproduce the problem.
